File: src/Other/Internals/ViewSourceButtonInfo.vb

Namespace Global.OpenSilver.Samples.Showcase
    Public Class ViewSourceButtonInfo
        Public Sub New()
        End Sub

        Public Sub New(relativePath As String, fileName As String)
            Me.RelativePath = relativePath
            Me.FileName = fileName
        End Sub

        Public Property FileName As String

        Public Property RelativePath As String

        Public Property Branch As String = "develop"

        Public Property Repository As String = "OpenSilver.Samples.Showcase"

        Public Property Owner As String = "OpenSilver"

        Public Property TabHeader As String

        Public Function GetHeader() As String
            Return If(Not String.IsNullOrEmpty(TabHeader), TabHeader, FileName)
        End Function

        Public Function GetAbsoluteUrl() As String
            Return $"https://github.com/{Owner}/{Repository}/blob/{Branch}/{RelativePath}/{FileName}"
        End Function
    End Class
End Namespace

File: src/OpenSilver.Showcase.Shared/ViewSource/ViewSourceButtonInfo.cs
namespace OpenSilver.Showcase
{
    public class ViewSourceButtonInfo
    {
        public ViewSourceButtonInfo() { }

        public ViewSourceButtonInfo(string relativePath, string fileName)
        {
            RelativePath = relativePath;
            FileName = fileName;
        }

        public string FileName { get; set; }

        public string RelativePath { get; set; }

        public string Commit { get; set; } = "de48e63f0e15f4edbfcf21e5d417af303bdcc2da";

        public string Repository { get; set; } = "OpenSilver.Samples.Showcase";

        public string Owner { get; set; } = "OpenSilver";

        public string TabHeader { get; set; }

        public string Fragment { get; set; }

        public string Notes { get; set; }

        public string GetHeader() => !string.IsNullOrEmpty(TabHeader) ? TabHeader : FileName;

        public string GetAbsoluteUrl() => $"https://github.com/{Owner}/{Repository}/blob/{Commit}/{RelativePath}/{FileName}{Fragment}";
    }
}
